bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/TypeChecker.java

```diff
@@ -1386,6 +1391,22 @@ private static boolean checkIsRecordType(MapValue sourceRecordValue, BRecordType
         return true;
     }
 
+    private static boolean containsInvalidNeverField(String fieldName, Field sourceField, MapValue sourceRecordValue) {
+        if (sourceField != null) {
```

Comment:
As the method logic is changed when addressing other suggestions, I think we don't need to inverse the if condition. Because the else part is more complex than the if part.